lib/components/primitive-components/Group/Group_doInitialSimulationSpiceEngineRender.ts

```diff
@@ -29,8 +29,20 @@ export function Group_doInitialSimulationSpiceEngineRender(group: Group<any>) {
   // Convert circuit JSON to SPICE string
   let spiceString: string
   try {
-    const spiceNetlist = circuitJsonToSpice(circuitJson)
-    spiceString = spiceNetlist.toSpiceString()
+    // Get simulation parameters from AnalogSimulation component
+    const simulationProps = analogSims[0]?._parsedProps
+    const duration = simulationProps?.duration ?? "10ms"
+    const timePerStep = simulationProps?.timePerStep ?? "1us"
+
+    let spiceNetlistStr = circuitJsonToSpice(circuitJson).toSpiceString()
+
+    if (!spiceNetlistStr.includes(".tran")) {
+      spiceNetlistStr = spiceNetlistStr.replace(
+        /\.END/i,
+        `.tran ${timePerStep} ${duration}\n.END`,
```

Comment:
this seems hacky to me, why are we replacing? circuitJsonToSpice should be handling this